### ISSUE TITLE
APM Conflict with Core ANE

### ISSUE THREAD

**Issue by apofis1969** [airsdk/apm#170]
Hi,

I tried to do the ANE updates today, but get this conflict message. Is the rollout not ready until now?
✓ Found package: com.distriqt.Application@6.12.0   
✓ Found package: com.distriqt.Core@7.0.2   
✓ Found package: com.distriqt.playservices.AppSet@16.1.201 
CONFLICT :: fatal error : found [1] conflicting packages
CONFLICT :: com.distriqt.Core
CONFLICT :: ├── 7.0.2 required by: com.distriqt.Application@6.12.0
CONFLICT :: └── 6.4.8 required by: com.distriqt.Application@6.11.0
install ::

**Comment by marchbold** [comment]
That looks odd, there shouldn't be a conflict between the same extension. 

Can you post your `project.apm` and lock file for me?

**Comment by apofis1969** [comment]
Hi, I got it working. I deleted all APM Files and created a new project.
But there is something I do not understand:
Until yesterday I used the AIR_HOME path to the overlaid AIRSDK_Flex_MacOS, which I use to build the app, and it worked.
But suddenly I got this error message:
`Users/xxx/APM/apm_1.3.0/apm: line 33:   633 Killed: 9               "${ADL_DIR}"adl -profile extendedDesktop -nodebug -multi-instance -cmd "${SCRIPT_DIR}/apm.xml" -- -workingdir "$( pwd )" -appdir "${SCRIPT_DIR}" -airdir "${AIR_DIR}" -uname "${UNAME}" "$@"`
So I changed the AIR_HOME path to AIRSDK_MacOS and it is working. Strange... 
I am using Monterey 12.6.2, SDK 50.1.1.2

**Comment by marchbold** [comment]
@apofis1969 Did you change anything else or did it just stop working all of a sudden? Was there any more to the error, seems a bit incomplete?

**Comment by apofis1969** [comment]
@marchbold hi, no stopped working suddenly. These are the error messages I get. Not more or less. It is strange. It happened after the update to macOS 12.6.2
I really can’t understand this. 
if I try to install an ANE I get the same error message with a different error at the line.

**Comment by marchbold** [comment]
Hmm very odd. Likely this is something to do with the permissions/quarantine status on `adl` 

Could you try removing the quarantine status flags:

```
xattr -r -d com.apple.quarantine SDK_Folder
```

**Comment by apofis1969** [comment]
Hi,

I still get the error message. I made apm -version with the flex sdk: `/Users/xxxx/a-exports/APM/apm_1.3.0/apm: line 33:   630 Killed: 9               "${ADL_DIR}"adl -profile extendedDesktop -nodebug -multi-instance -cmd "${SCRIPT_DIR}/apm.xml" -- -workingdir "$( pwd )" -appdir "${SCRIPT_DIR}" -airdir "${AIR_DIR}" -uname "${UNAME}" "$@"`
Attached I send you the full error log which I had been asked to send to apple. The error message at the report is, that "all quit unexpectedly"
Also I send attached the error messages doing `xattr -r -d com.apple.quarantine SDK_Folder`
No idea what is going on here.
[apm-error.txt](https://github.com/airsdk/apm/files/10662490/apm-error.txt)
[quarantine.txt](https://github.com/airsdk/apm/files/10662491/quarantine.txt)

**Comment by marchbold** [comment]
@apofis1969 Can you run `adl` on its own from the command line?

@ajwfrost Do you have any ideas here? Seems adl is crashing for some reason?

**Comment by ajwfrost** [comment]
Seems to be due to code signing? From that Apple report:
```
Exception Type:  EXC_CRASH (SIGKILL (Code Signature Invalid))
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: CODESIGNING 1 
```

Worth checking the outputs of these perhaps?
```
codesign -v -vvv "${ADL_DIR}adl"
codesign -v -vvv "${ADL_DIR}../runtimes/air/mac/Adobe AIR.framework"
```

**Comment by apofis1969** [comment]
@marchbold can you tell me which command? adl ... then I will try this and tell you the result

**Comment by marchbold** [comment]
Just try `adl` by itself, should print out some version / usage information and check the codesigning as Andrew suggested?

**Comment by apofis1969** [comment]
this happens when I only try adl
`Translated Report (Full Report Below)
-------------------------------------

Incident Identifier: ED48C4D0-BA14-4DC5-88FC-38D7507B700C
CrashReporter Key:   7D59C6CE-92BE-AC34-E246-46BEB0F2BC07
Hardware Model:      MacBookPro11,5
Process:             adl [599]
Path:                /Users/USER/Documents/*/adl
Identifier:          com.adobe.air.ADL
Version:             32.0 (32.0.0.144)
Code Type:           X86-64 (Native)
Role:                Unspecified
Parent Process:      zsh [565]
Coalition:           com.apple.Terminal [665]
Responsible Process: Terminal [558]

Date/Time:           2023-02-07 21:44:09.5345 -0300
Launch Time:         2023-02-07 21:44:09.4063 -0300
OS Version:          macOS 12.6.2 (21G320)
Release Type:        User
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL (Code Signature Invalid))
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: CODESIGNING 1 

Highlighted by Thread:  0

Backtrace not available

No thread state (register information) available

Binary Images:
Binary images description not available

Error Formulating Crash Report:
_dyld_process_info_create failed with 6
dyld_process_snapshot_get_shared_cache failed
Failed to create CSSymbolicatorRef - corpse still valid ¯\_(ツ)_/¯

EOF

-----------
Full Report
-----------

{"app_name":"adl","timestamp":"2023-02-07 21:44:11.00 -0300","app_version":"32.0","slice_uuid":"45f0d9ad-d293-35e7-9cb1-d84a148ef46d","build_version":"32.0.0.144","platform":0,"bundleID":"com.adobe.air.ADL","share_with_app_devs":0,"is_first_party":1,"bug_type":"309","os_version":"macOS 12.6.2 (21G320)","incident_id":"ED48C4D0-BA14-4DC5-88FC-38D7507B700C","name":"adl"}
{
  "uptime" : 63,
  "procLaunch" : "2023-02-07 21:44:09.4063 -0300",
  "procRole" : "Unspecified",
  "version" : 2,
  "userID" : 502,
  "deployVersion" : 210,
  "modelCode" : "MacBookPro11,5",
  "procStartAbsTime" : 63116602695,
  "coalitionID" : 665,
  "osVersion" : {
    "train" : "macOS 12.6.2",
    "build" : "21G320",
    "releaseType" : "User"
  },
  "captureTime" : "2023-02-07 21:44:09.5345 -0300",
  "incident" : "ED48C4D0-BA14-4DC5-88FC-38D7507B700C",
  "bug_type" : "309",
  "pid" : 599,
  "procExitAbsTime" : 63244020313,
  "cpuType" : "X86-64",
  "procName" : "adl",
  "procPath" : "\/Users\/USER\/Documents\/*\/adl",
  "bundleInfo" : {"CFBundleShortVersionString":"32.0","CFBundleVersion":"32.0.0.144","CFBundleIdentifier":"com.adobe.air.ADL"},
  "parentProc" : "zsh",
  "parentPid" : 565,
  "coalitionName" : "com.apple.Terminal",
  "crashReporterKey" : "7D59C6CE-92BE-AC34-E246-46BEB0F2BC07",
  "responsiblePid" : 558,
  "responsibleProc" : "Terminal",
  "sip" : "enabled",
  "isCorpse" : 1,
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGKILL (Code Signature Invalid)"},
  "termination" : {"namespace":"CODESIGNING","flags":66,"code":1},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "legacyInfo" : {
  "threadHighlighted" : 0
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "6112dda2fc54bc3389840642",
      "factorPackIds" : {
        "SIRI_DICTATION_ASSETS" : "620aec83b02b354d3afd2f50"
      },
      "deploymentId" : 240000145
    },
    {
      "rolloutId" : "60186475825c62000ccf5450",
      "factorPackIds" : {

      },
      "deploymentId" : 240000026
    }
  ],
  "experiments" : [

  ]
},
  "reportNotes" : [
  "_dyld_process_info_create failed with 6",
  "dyld_process_snapshot_get_shared_cache failed",
  "Failed to create CSSymbolicatorRef - corpse still valid ¯\\_(ツ)_\/¯"
]
}

Model: MacBookPro11,5, BootROM 476.0.0.0.0, 4 processors, Quad-Core Intel Core i7, 2,8 GHz, 16 GB, SMC 2.30f2
Graphics: Intel Iris Pro, Intel Iris Pro, Built-In
Graphics: AMD Radeon R9 M370X, AMD Radeon R9 M370X, PCIe, 2 GB
Display: Color LCD, 2880 x 1800 Retina, Main, MirrorOff, Online
Display: PHL 288P6L, 3840 x 2160 (2160p/4K UHD 1 - Ultra High Definition), MirrorOff, Online
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533642465238412D50422020
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533642465238412D50422020
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x152), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1710.4)
AirPort: 
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD SM1024G, 1 TB
USB Device: USB30Bus
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Bluetooth USB Host Controller
USB Device: HD Pro Webcam C920
Thunderbolt Bus: MacBook Pro, Apple Inc., 27.1
`

**Comment by ajwfrost** [comment]
So that's Adobe's build of ADL ("build_version":"32.0.0.144"), I doubt they'd have done all the code signing stuff that's necessary for the latest macOS versions.

There may be something up with your path variables or similar, that's making it pick up this ADL version rather than the ones you'd mentioned earlier. I think if you make sure you've got the PATH variable set up so that when you run "adl" you get a working app that outputs a version 33.x or 50.x then it may help?

thanks

**Comment by ajwfrost** [comment]
Okay so that version thing we can ignore:

> There the 32.0.0.144 comes from, maybe this is the bug. It is in /Users/xxxx/a-airsdk/SDKS/sdk50.1.1.2/sdk50-1-1-2/bin/Contents/Info.plist

That file isn't part of the SDK any more so must be left over from an earlier merge...


So then it's just the signing thing...
The path looks like it will have both:
`/Users/xxxx/a-airsdk/SDKS/sdk50.1.1.2/sdk50-1-1-2/bin`
and
`/Users/xxxx/a-airsdk/SDKS/sdk50.1.1.2/AIRSDK_MacOS/bin`
 in it, but the first of these is likely to be picked up first. So if you have a terminal window, you can check that with:
```
which adl
```
and then using that part, try:
```
codesign -vvv /path/bin/adl
codesign -dvv /path/bin/adl
```

Actually, it may just be as simple as deleting that "Contents" folder with the info.plist in there, that's likely to be confusing the OS...

thanks

**Comment by apofis1969** [comment]
@ajwfrost it was definitely the Contents Folder. I removed and now it is working again. I also did the code signing. But even with the signed adl it did not work until I removed the Content Folder

**Comment by ajwfrost** [comment]
Great, thanks ... so yes, I think that was why we took out that Contents/Info.plist in the first place. Adobe had added it because ADL needed some permission/workaround to do http accesses or similar, but Apple have changed how they process code signatures and the permissions are now embedded into the binary itself. So that file just would have confused the system....

So -> the original error in the description (apm conflict) seems to have been resolved by a refresh, and the subsequent issue with ADT being killed is hopefully sorted by getting rid of that old file/folder -> I would say we should close this one now?

thanks
